### DISCUSSION TITLE
Add story `parameters` to the `item` via the `renderLabel` function

### DISCUSSION THREAD

**Discussion by mririgoyen** [storybookjs/storybook#24022] [Ideas]
### Is your feature request related to a problem? Please describe.

I need to be able to conditionally add status icons to my Component names. I thought I could easily do this utilizing the `renderLabel` function for the sidebar, however the `item` that is available is quite limited in the information that it has available. According to the Storybook docs, ["Parameters are a set of static, named metadata about a story, typically used to control the behavior of Storybook features and addons."](https://storybook.js.org/docs/react/writing-stories/parameters) However, I cannot access these in the addon config to actually utilize them.

### Describe the solution you'd like

I would like to be able to access the parameters for each `item` in the `renderLabel` function, so that I can make rendering decisions on the title. For example:

```jsx
import { addons } from '@storybook/manager-api';
import Icon from '@mdi/react';
import { mdiAccountCircle, mdiServer } from '@mdi/js';

addons.setConfig({
  sidebar: {
    // Access "parameters" as part of the item here
    renderLabel: ({ name, parameters, type }) => {
      // Only add icons to components
      if (type !== 'component') {
        return name;
      }
      
      // Get my custom parameter from the story metadata
      const { componentType } = parameters;
      
      // Only add icons if a componentType is specified
      if (!['client', 'server'].includes(componentType)) {
        return name;
      }

      const icon = componentType === 'server' ? mdiServer : mdiAccountCircle;
      return <>{name} <Icon path={icon} size={1} />;
    },
  },
});
```

### Describe alternatives you've considered

I've tried to use a custom `storyIndexers` function in `main.ts`, but no matter how I configure it, it never matches any files.

### Are you able to assist to bring the feature to reality?

no

**Comment by awreese** [comment]
I second this request.

Interestingly enough it seems that there *should*/*could* be a `parameters` property for *certain* item types since there's an interface/type declared with it. I don't see `parameters` on `API_ComponentEntry` though, which is what it seems @mririgoyen is asking for.

https://github.com/storybookjs/storybook/blob/2ebbeed44d29be79b8c778e81f2cba4e9047d8b7/code/core/src/types/modules/api-stories.ts#L26-L65

But in the current code I have (SB v8) the `parameters` property is not passed through and just simply undefined.

**Reply by matths** [reply]
I managed this right now, my .storybook/manager.tsx looks like this:

```
import React from 'react';
import type { API_HashEntry } from 'storybook/internal/types';
import { addons } from '@storybook/manager-api';

addons.setConfig({
  sidebar: {
    renderLabel: (item: API_HashEntry, api: any) => {
      const parameters = api.getParameters(item.id);
      if (parameters?.class) {
        return <span className={parameters.class}>{item.name}</span>;
      }
      return item.name;
    },
  },
});
```

So I can add a 'class' property to every Story Metadata parameter object, and it is mapped to the leaf nodes in my storybook explorer tree. I add a style tag to the .storybook/manager-head.html

Best,
Matthias

**Reply by Jintus** [reply]
This solution works only for leaf which are currently opened. If not getParameters return null

**Reply by matths** [reply]
you are right, sorry for the confusion. Any other ideas then?
I played some time with experimental_indexers inside StorybookConfig (.storybook/main.ts), but without luck. It is hard to get Meta data from a custom svelte story at that point in time.

**Reply by Jintus** [reply]
Yeah I do not find solutions using only args passed to the function.

For my use case, tags could have been the preferred solution. But due to static analysis, in story, tags property cannot be populated with imported variable like `tags: ['value', ...importedConst]`

So my **not ideally** solution is to generate a file which list all story ids which should have a tag, and in my render label function I compare id with those values. If you have any other idea I'm all ears !

**Comment by shilman** [comment]
Unfortunately this is not feasible since stories are _dynamically_ loaded, but parameters while static/unchanging, cannot be _statically_ analyzed. Therefore, we only know the values of the parameters when we load the story. But the sidebar is based on statically analyzable data.

As a workaround, please consider using [tags](https://storybook.js.org/docs/writing-stories/tags), which **are** statically analyzable and are available as part of the `item` that's passed into `renderLabel`. For a full example of this approach, please see https://github.com/Sidnioulz/storybook-addon-tag-badges cc @Sidnioulz
